runtime/parser2/declaration.go

```diff
@@ -441,30 +441,38 @@ func parseImportDeclaration(p *parser) *ast.ImportDeclaration {
 
 			case lexer.TokenIdentifier:
 
-				if p.current.Value == keywordFrom {
+				if p.current.Value != keywordFrom {
+					identifier := tokenToIdentifier(p.current)
+					identifiers = append(identifiers, identifier)
 
-					if !expectCommaOrFrom {
-						panic(fmt.Errorf(
-							"expected %s, got keyword %q",
-							lexer.TokenIdentifier,
-							p.current.Value,
-						))
-					}
+					expectCommaOrFrom = true
+					break
+				}
 
+				if expectCommaOrFrom {
 					atEnd = true
 
 					// Skip the `from` keyword
 					p.next()
 					p.skipSpaceAndComments(true)
 
 					parseLocation()
-				} else {
-					identifier := tokenToIdentifier(p.current)
-					identifiers = append(identifiers, identifier)
+					break
+				}
 
-					expectCommaOrFrom = true
+				if !isCurrentTokenAnImportedIdentifier(p) {
+					panic(fmt.Errorf(
+						"expected %s, got keyword %q",
+						lexer.TokenIdentifier,
+						p.current.Value,
+					))
 				}
 
+				identifier := tokenToIdentifier(p.current)
+				identifiers = append(identifiers, identifier)
+
+				expectCommaOrFrom = true
```

Comment:
Looks like this code is now duplicated from above (the code in the `if p.current.Value != keywordFrom {` branch)